cmd/release-controller/bugzilla.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	v1 "github.com/openshift/api/image/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+)
+
+// syncBugzilla checks whether fixed bugs in a release had their
+// PR reviewed and approved by the QA contact for the bug
+func (c *Controller) syncBugzilla(key queueKey) error {
+	defer func() {
+		err := recover()
+		panic(err)
+	}()
+
+	release, err := c.loadReleaseForSync(key.namespace, key.name)
+	if err != nil || release == nil {
+		return err
+	}
+
+	// check if verifyBugs publish step is enabled for release
+	var verifyBugs *PublishVerifyBugs
+	for _, publishType := range release.Config.Publish {
+		if publishType.Disabled {
+			continue
+		}
+		switch {
+		case publishType.VerifyBugs != nil:
+			verifyBugs = publishType.VerifyBugs
+			break
+		}
+	}
+	if verifyBugs == nil {
+		return nil
+	}
+
+	glog.V(4).Infof("Verifying fixed bugs in %s", release.Config.Name)
+
+	// get accepted tags
+	acceptedTags := sortedRawReleaseTags(release, releasePhaseAccepted)
+	// We need at least 2 accepted releases to compare
+	if len(acceptedTags) < 2 {
+		return nil
+	}
+	var errs []error
+	// get latest accepted tag that has not been bugzilla verified
+	var tag, prevTag *v1.TagReference
+	for index, currTag := range acceptedTags {
+		if tag.Annotations[releaseAnnotationBugsVerified] == "true" {
+			// if we've already checked the latest tag, return
+			if index == 0 {
+				return nil
+			}
+			prevTag = currTag
+			tag = acceptedTags[index-1]
+			break
+		}
+	}
+
+	verifyComplete := false
+	if bugs, err := c.releaseInfo.Bugs(prevTag.Name, tag.Name); err != nil {
+		errs = append(errs, fmt.Errorf("Unable to generate changelog from %s to %s: %v", prevTag.Name, tag.Name, err))
```

Comment:
If you get an error you should be returning here for a retry, not continuing.  In general, in go (in any of our codebases) always exit early.  In the release controller case (and other controllers) the pattern is that the code calling this function (a sync or "process one item" function) will handle errors and retries, so it's better for you to exit early.  The queue that invokes this function has error retry and backoff function, and if you can't retrieve bugs you just want to retry later (vs moving on).

So this should just be:

```
bugs, err := c.releaseInfo.Bugs(...)
if err != nil {
  return fmt.Errorf("unable to generate changelog ....", ...)
}
if errs := c.bugzillaVerifier.VerifyBugs(bugs); len(errs) > 0 {
  return utilerrors.NewAggregate(errs)
}

release.Target.DeepCopy()
tagToBeUpdated := findTagReference(target, tag.Name)
...
if _, err := c.imageClient.ImageStreams(target.Namespace).Update(target); err != nil {
  return err
}
...
return nil
```